content/spin/javascript-components.md

```diff
@@ -0,0 +1,281 @@
+title = "Building Spin Components in JavaScript"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+enable_shortcodes = true
+---
+
+With JavaScript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features. 
+
+> This guide assumes you are familiar with the Javascript programming language,
+> but if you are just getting started, be sure to check [the MDN guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide).
+
+> All examples from this page can be found in [the Spin repository on GitHub](https://github.com/fermyon/spin-js-sdk/tree/main/examples).
+
+In order to compile JavaScript programs to Spin components, you also need to install a spin plugin `js2wasm` using the following command:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin plugin update
```

Comment:
Is this command always needed?

Reply:
This is currently needed if someone has already installed some plugin. In most cases no but without it there is a possibility that the next command will throw an error saying manifest for js2wasm not found.